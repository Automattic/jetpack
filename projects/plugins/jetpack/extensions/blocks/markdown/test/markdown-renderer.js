/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import React from 'react';

/**
 * Internal dependencies
 */
import { source } from './fixtures/source'
import MarkdownRenderer from '../renderer';

describe( 'MarkdownRenderer', () => {
	test( 'renders markdown to HTML as expected', () => {
		expect( shallow( <MarkdownRenderer className='markdown' source={ source } /> ) ).toMatchSnapshot();
	} );
} );
