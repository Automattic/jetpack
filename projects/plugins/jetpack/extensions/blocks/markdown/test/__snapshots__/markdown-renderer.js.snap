// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownRenderer renders markdown to HTML as expected 1`] = `
<RawHTML
  className="markdown"
  onClick={[Function]}
>
  &lt;h1&gt;Heading&lt;/h1&gt;
&lt;h2&gt;2nd Heading&lt;/h2&gt;
&lt;ul&gt;
&lt;li&gt;List 1&lt;/li&gt;
&lt;li&gt;List 1&lt;/li&gt;
&lt;/ul&gt;
&lt;ul&gt;
&lt;li&gt;List 2&lt;/li&gt;
&lt;li&gt;List 2&lt;/li&gt;
&lt;/ul&gt;
&lt;ul&gt;
&lt;li&gt;List 3&lt;/li&gt;
&lt;li&gt;List 3&lt;/li&gt;
&lt;/ul&gt;
&lt;ol&gt;
&lt;li&gt;Red&lt;/li&gt;
&lt;li&gt;Green&lt;/li&gt;
&lt;li&gt;Blue&lt;/li&gt;
&lt;/ol&gt;
&lt;ul&gt;
&lt;li&gt;
&lt;p&gt;A list item.&lt;/p&gt;
&lt;p&gt;With multiple paragraphs.&lt;/p&gt;
&lt;/li&gt;
&lt;li&gt;
&lt;p&gt;Another item in the list.&lt;/p&gt;
&lt;/li&gt;
&lt;/ul&gt;
&lt;p&gt;&lt;em&gt;em&lt;/em&gt;
&lt;em&gt;em&lt;/em&gt;
&lt;strong&gt;strong&lt;/strong&gt;
&lt;strong&gt;strong&lt;/strong&gt;
&lt;em&gt;&lt;strong&gt;em strong&lt;/strong&gt;&lt;/em&gt;&lt;/p&gt;
&lt;p&gt;*Literal asterisks*&lt;/p&gt;
&lt;p&gt;Link to &lt;a href="https://wordpress.com"&gt;WordPress&lt;/a&gt; and &lt;a href="https://jetpack.com/"&gt;https://jetpack.com/&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;email me: &lt;a href="mailto:address@example.com"&gt;address@example.com&lt;/a&gt;&lt;/p&gt;
&lt;p&gt;Inline &lt;code&gt;code&lt;/code&gt; here.&lt;/p&gt;
&lt;pre&gt;&lt;code&gt;Block of code with backticks.
&lt;/code&gt;&lt;/pre&gt;
&lt;pre&gt;&lt;code&gt;Block of code prefixed by four spaces
&lt;/code&gt;&lt;/pre&gt;
&lt;blockquote&gt;
&lt;p&gt;a blockquote.&lt;/p&gt;
&lt;p&gt;2nd paragraph in the blockquote.&lt;/p&gt;
&lt;h2&gt;H2 in a blockquote&lt;/h2&gt;
&lt;/blockquote&gt;
&lt;p&gt;A bunch of horizontal rules:&lt;/p&gt;
&lt;hr&gt;
&lt;hr&gt;
&lt;hr&gt;
&lt;hr&gt;
&lt;hr&gt;
&lt;p&gt;ðŸ‘‹&lt;/p&gt;

</RawHTML>
`;
